File: offer/45.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var minNumber = function(nums) {
    return nums.sort(sortNums).join('')
};

const sortNums = (a, b) => {
    a = String(a)
    b = String(b)
    if(a[0] !== b[0]) return a[0] - b[0]
    return Number(a + b) - Number(b + a)
}

File: offer/45/45.go
package offer

import (
	"fmt"
	"sort"
	"strings"
)

func minNumber(nums []int) string {
	sort.Slice(nums, func(i, j int) bool {
		x := fmt.Sprintf("%d%d", nums[i], nums[j])
		y := fmt.Sprintf("%d%d", nums[j], nums[i])
		return x < y
	})

	var b strings.Builder
	for i := 0; i < len(nums); i++ {
		// b.WriteString(strconv.Itoa(nums[i]))
		b.WriteString(fmt.Sprintf("%d", nums[i]))
	}
	return b.String()

	// res := ""
	// for i:=0; i<len(nums); i++ {
	//     res += fmt.Sprintf("%d", nums[i])
	// }
	// return res
}
